src/components/Drawer/Navigation/Navigation.tsx

```diff
@@ -0,0 +1,65 @@
+/** @jsx jsx */
+import { jsx } from '@emotion/core';
+import React, { useCallback, useLayoutEffect, useState } from 'react';
+import { useLocation } from 'react-router-dom';
+import { navigationContainerStyle } from './Navigation.style';
+import { Props } from '../Drawer';
+import MenuItem from './MenuItem/MenuItem';
+
+const Navigation: React.FC<Props> = ({ menuItems }) => {
+  const location = useLocation();
+  const [currentMenuItem, setCurrentMenuItem] = useState<string>();
+  const [openMenuItems, setOpenMenuItems] = useState<string[]>([]); // we identify open menuitems by their url
+
+  const toggleMenuItem = useCallback((newUrl: string): void => {
+    setOpenMenuItems(currentOpenMenuItems => {
+      // check if the newly selected Menuitem is currently open
+      const menuItemIndex = currentOpenMenuItems.findIndex(menuItemUrl => menuItemUrl === newUrl);
+      if (menuItemIndex === -1) {
+        // The newly selected item is not currently open, add to selection
+        return [...currentOpenMenuItems, newUrl];
+      } else {
+        // The newly selected item is already open
+        const newOpenMenuItems = [...currentOpenMenuItems];
+        newOpenMenuItems.splice(menuItemIndex, 1);
+
+        return newOpenMenuItems;
+      }
+    });
+  }, []);
+
+  // we need to show an indication on the currently selected option
+  // we get that information from the current URL and then comparing it with the urls of each menuItem
+  useLayoutEffect(() => {
+    menuItems.some(menuItem => {
```

Comment:
use of `.some` indicates that will return something in a const for something else. If this is just a loop `.forEach` could work just fine